extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -24,123 +28,73 @@ const { naturalizeDataset, denaturalizeDataset } = DicomMetaDictionary;
-      if (dicomWebConfig.onConfiguration && typeof dicomWebConfig.onConfiguration === 'function') {
-        dicomWebConfig = dicomWebConfig.onConfiguration(dicomWebConfig, {
-          params,
-          query,
-        });
-      }
-
-      dicomWebConfigCopy = JSON.parse(JSON.stringify(dicomWebConfig));
-
-      getAuthrorizationHeader = () => {
-        const xhrRequestHeaders = {};
-        const authHeaders = userAuthenticationService.getAuthorizationHeader();
-        if (authHeaders && authHeaders.Authorization) {
-          xhrRequestHeaders.Authorization = authHeaders.Authorization;
-        }
-        return xhrRequestHeaders;
-      };
-
-      generateWadoHeader = () => {
-        let authorizationHeader = getAuthrorizationHeader();
-        //Generate accept header depending on config params
-        let formattedAcceptHeader = utils.generateAcceptHeader(
-          dicomWebConfig.acceptHeader,
-          dicomWebConfig.requestTransferSyntaxUID,
-          dicomWebConfig.omitQuotationForMultipartRequest
-        );
-
-        return {
-          ...authorizationHeader,
-          Accept: formattedAcceptHeader,
-        };
-      };
-
-      qidoConfig = {
-        url: dicomWebConfig.qidoRoot,
-        staticWado: dicomWebConfig.staticWado,
-        singlepart: dicomWebConfig.singlepart,
-        headers: userAuthenticationService.getAuthorizationHeader(),
-        errorInterceptor: errorHandler.getHTTPErrorHandler(),
-      };
-
-      wadoConfig = {
-        url: dicomWebConfig.wadoRoot,
-        staticWado: dicomWebConfig.staticWado,
-        singlepart: dicomWebConfig.singlepart,
-        headers: userAuthenticationService.getAuthorizationHeader(),
-        errorInterceptor: errorHandler.getHTTPErrorHandler(),
-      };
-
-      // TODO -> Two clients sucks, but its better than 1000.
-      // TODO -> We'll need to merge auth later.
-      qidoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(qidoConfig)
-        : new api.DICOMwebClient(qidoConfig);
-
-      wadoDicomWebClient = dicomWebConfig.staticWado
-        ? new StaticWadoClient(wadoConfig)
-        : new api.DICOMwebClient(wadoConfig);
+      dicomWebClientManager = new DicomWebClientManager({
+        params,
+        query,
+        configs: dicomWebConfig,
+        userAuthenticationService,
+      });
     },
     query: {
       studies: {
         mapParams: mapParams.bind(),
         search: async function (origParams) {
-          qidoDicomWebClient.headers = getAuthrorizationHeader();
-          const { studyInstanceUid, seriesInstanceUid, ...mappedParams } =
-            mapParams(origParams, {
-              supportsFuzzyMatching: dicomWebConfig.supportsFuzzyMatching,
-              supportsWildcard: dicomWebConfig.supportsWildcard,
-            }) || {};
-
-          const results = await qidoSearch(qidoDicomWebClient, undefined, undefined, mappedParams);
+          const clients = dicomWebClientManager.getQidoClients();
```

Comment:
Just separate out the clients into separate configs, and lookup the global config with the given name.  That way you can have multiple top level query entry points, and multiple users of those.  That is:
RetrieveAE => query locaiton 1 with study data for some specific purpose
RetrieveAE 2 - global study data with global study query
AE 2 points to AE 1 for AE1 specific data, but otherwise AE 1 points to AE 2 for everything imaging related.
There are lots of situations like this.   A true merge AE is a merge of the study level query results, and should be it's own data source type that is JUST merge.